src/thor/triplegbuilder.cc

```diff
@@ -1320,27 +1315,11 @@ TripLegBuilder::Build(const AttributesController& controller,
       trip_node->set_elapsed_time(elapsedtime);
     }
 
-    double edge_pct = 1.0;
-    if (trim_duration) {
-      // scenario 1: edge is the first edge
-      //                   trip_node
-      // edge start ==========O========== edge end
-      //                      |__________|
-      //                       duration
-      if (edge_itr == path_begin) {
-        edge_pct = 1.f - start_pct;
-      }
-      // scenario 2: edge is the last edge
-      //                   trip_node
-      // edge start ==========O========== edge end
-      //           |__________|
-      //             duration
-      else if (edge_itr == path_end - 1) {
-        edge_pct = end_pct;
-      }
-    }
     // Update elapsed time at the end of the edge, store this at the next node.
-    elapsedtime += edge_itr->elapsed_time * edge_pct;
+    elapsedtime = edge_itr->elapsed_time - trim_begin;
```

Comment:
same as trivial path above. basically we always remove the previous legs elapsed time from the map match elapsed time for the current leg. remember the elapsed time is cumulative. then below we also need to remove whatever partial time remains on the edge the leg ends on